lms/djangoapps/badges/api/tests.py

```diff
@@ -0,0 +1,234 @@
+    """
+    WILDCARD = False
+    CHECK_COURSE = False
+
+    def setUp(self, *args, **kwargs):
+        super(UserAssertionTestCase, self).setUp(*args, **kwargs)
+        self.course = CourseFactory.create()
+        self.user = UserFactory.create()
+        # Password defined by factory.
+        self.client.login(username=self.user.username, password='test')
+
+    def url(self):
+        """
+        Return the URL to look up the current user's assertions.
+        """
+        return '/api/badges/v1/assertions/user/{}/'.format(self.user.username)
+
+    def check_class_structure(self, badge_class, json_class):
+        """
+        Check a JSON response against a known badge class.
+        """
+        self.assertEqual(badge_class.issuing_component, json_class['issuing_component'])
+        self.assertEqual(badge_class.slug, json_class['slug'])
+        self.assertIn(badge_class.image.url, json_class['image_url'])
+        self.assertEqual(badge_class.description, json_class['description'])
+        self.assertEqual(badge_class.criteria, json_class['criteria'])
+        self.assertEqual(badge_class.course_id and unicode(badge_class.course_id), json_class['course_id'])
+
+    def check_assertion_structure(self, assertion, json_assertion):
+        """
+        Check a JSON response against a known assertion object.
+        """
+        self.assertEqual(assertion.image_url, json_assertion['image_url'])
+        self.assertEqual(assertion.assertion_url, json_assertion['assertion_url'])
+        self.assertTrue(json_assertion['user'].startswith('http'))
+        self.check_class_structure(assertion.badge_class, json_assertion['badge_class'])
+
+    def get_course_id(self, badge_class):
+        """
+        Used for tests which may need to test for a course_id or a wildcard.
+        """
+        if self.WILDCARD:
+            return '*'
+        else:
+            return unicode(badge_class.course_id)
+
+    def create_badge_class(self, **kwargs):
+        """
+        Create a badge class, using a course id if it's relevant to the URL pattern.
+        """
+        if self.CHECK_COURSE:
+            return RandomBadgeClassFactory.create(course_id=self.course.location.course_key, **kwargs)
+        return RandomBadgeClassFactory.create(**kwargs)
+
+    def get_qs_args(self, badge_class):
+        """
+        Get a dictionary to be serialized into querystring params based on class settings.
+        """
+        qs_args = {
+            'issuing_component': badge_class.issuing_component,
+            'slug': badge_class.slug,
+        }
+        if self.CHECK_COURSE:
+            qs_args['course_id'] = self.get_course_id(badge_class)
+        return qs_args
+
+
+class TestUserBadgeAssertions(UserAssertionTestCase):
+    """
+    Test the general badge assertions retrieval view.
+    """
+
+    def test_get_assertions(self):
+        """
+        Verify we can get all of a user's badge assertions.
+        """
+        for dummy in range(3):
```

Comment:
The first character listed in that link is `_`.  So that's what I'm referring to.  And since we want to distinguish between the `_` that refers to `ugettext`, I usually use double underscore instead.

But yeah, `dummy` also works as an old-style convention.